src/librustc_mir/transform/validate.rs

```diff
@@ -57,24 +91,38 @@ impl<'a, 'tcx> Visitor<'tcx> for TypeChecker<'a, 'tcx> {
         self.super_operand(operand, location);
     }
 
-    fn visit_statement(&mut self, statement: &Statement<'tcx>, location: Location) {
-        // The sides of an assignment must not alias. Currently this just checks whether the places
-        // are identical.
-        if let StatementKind::Assign(box (dest, rvalue)) = &statement.kind {
-            match rvalue {
-                Rvalue::Use(Operand::Copy(src) | Operand::Move(src)) => {
-                    if dest == src {
-                        self.fail(
-                            DUMMY_SP,
-                            format!(
-                                "encountered `Assign` statement with overlapping memory at {:?}",
-                                location
```

Comment:
@jonas-schievink I am not sure why you printed the `Location` here when we could have a `Span` instead.

Alternatively, maybe `fail` should take a `Location` so that we can always print it?

Reply:
The `Location` is fairly useful when trying to find the offending statement in the MIR dump. Sounds like a good idea to always print it.

Reply:
Okay, I added that in https://github.com/rust-lang/rust/pull/72810.